editor/script_editor_debugger.cpp

```diff
@@ -399,55 +425,57 @@ void ScriptEditorDebugger::_parse_message(const String &p_msg, const Array &p_da
 		le_set->set_disabled(false);
 	} else if (p_msg == "message:inspect_object") {
 
+		ScriptEditorDebuggerInspectedObject *debugObj = NULL;
+
 		ObjectID id = p_data[0];
 		String type = p_data[1];
-		Variant path = p_data[2]; //what to do yet, i don't  know
-		int prop_count = p_data[3];
+		Array properties = p_data[2];
 
-		int idx = 4;
-
-		if (inspected_object->last_edited_id != id) {
-			inspected_object->prop_list.clear();
-			inspected_object->prop_values.clear();
+		bool is_new_object = false;
+		if (remote_objects.has(id)) {
+			debugObj = remote_objects[id];
+		} else {
+			debugObj = memnew(ScriptEditorDebuggerInspectedObject);
+			debugObj->remote_object_id = id;
+			debugObj->type_name = type;
+			remote_objects[id] = debugObj;
+			is_new_object = true;
+			debugObj->connect("value_edited", this, "_scene_tree_property_value_edited");
 		}
 
-		for (int i = 0; i < prop_count; i++) {
+		for (int i = 0; i < properties.size(); i++) {
```

Comment:
Please check my code for sending objects, it fixes a few bugs since 2.1 regarding objects being too big that may crash the debugger, so you may want to use it, or part of it.